ocs_ci/ocs/bucket_utils.py

```diff
@@ -2920,3 +2920,111 @@ def create_s3client_from_assume_role_creds(mcg_obj, assume_role_creds):
         aws_session_token=assumed_session_token,
     )
     return assumed_s3_resource.meta.client
+
+
+def put_bucket_versioning_via_awscli(
+    mcg_obj, awscli_pod, bucket_name, status="Enabled"
+):
+    """
+    Put bucket versioning using AWS CLI
+
+    Args:
+        mcg_obj (MCG): MCG object
+        awscli_pod (Pod): Pod object where AWS CLI is installed
+        bucket_name (str): Name of the bucket
+        status (str): Status of the versioning
+
+    """
+    awscli_pod.exec_cmd_on_pod(
+        command=craft_s3_command(
+            f"put-bucket-versioning --bucket {bucket_name} --versioning-configuration Status={status}",
+            mcg_obj=mcg_obj,
+            api=True,
+        )
+    )
+
+
+def upload_obj_versions(mcg_obj, awscli_pod, bucket_name, obj_key, amount=1, size="1M"):
+    """
+    Upload multiple random data versions to a given object key and return their ETag values
+
+    Args:
+        mcg_obj (MCG): MCG object
+        awscli_pod (Pod): Pod object where AWS CLI is installed
+        bucket_name (str): Name of the bucket
+        obj_key (str): Object key
+        amount (int): Number of versions to create
+        size (str): Size of the object. I.E 1M
+
+    Returns:
+        list: List of ETag values of versions in latest to oldest order
+    """
+    file_dir = f"/tmp/{uuid4()}"
+    awscli_pod.exec_cmd_on_pod(f"mkdir {file_dir}")
+
+    etags = []
+
+    for i in range(amount):
+        file_path = f"{file_dir}/{obj_key}_{i}"
+        awscli_pod.exec_cmd_on_pod(
+            command=f"dd if=/dev/urandom of={file_path} bs={size} count=1"
+        )
+        # Use debug and redirect it to stdout to get
+        # the uploaded object's ETag in the response
+        resp = awscli_pod.exec_cmd_on_pod(
+            command=craft_s3_command(
+                f"cp {file_path} s3://{bucket_name}/{obj_key} --debug 2>&1",
+                mcg_obj=mcg_obj,
+            ),
+            out_yaml_format=False,
+        )
+
+        # Parse the ETag from the response
+        # Filter the line containing the JSON
+        line = next(filter(lambda line: "ETag" in line, resp.splitlines()))
+        json_start = line.index("{")
+        json_str = line[json_start:]
+
+        # Fix quotes to read as dict
+        json_str = json_str.replace("'", '"')
+        json_str = json_str.replace('""', '"')
+
+        # Convert to dict and extract the ETag
+        new_etag = json.loads(json_str).get("ETag")
+
+        # Later versions should precede the older ones
+        # this achieves the expected order of versions
+        # we'd get from list-object-versions
+        etags.insert(0, new_etag)
+
+    return etags
```

Comment:
Why? We don't normally return the responses from AWSCLI commands done via the pod, and it would only add complexity.

Reply:
IMO, we could use same method on some other tests as well where we might need some other details like "versionId" instead of "etag". So returning entire response would be the great idea for reusability

Reply:
Ok I see your point - VersionId would be definitely useful in other tests. 

However, this method doesn't really get a normal response - it's parsing the ETags from the debug logs of the PutObject op, and I couldn't find VersionId in these logs. 

As a compromise, I modified `get_obj_versions` to return the response dicts instead of ETags in https://github.com/red-hat-storage/ocs-ci/pull/10957/commits/a9bba8a48b1d0dfdb70cd46155faf1bfeae47d69, so one could use that to get such MD if needed.